cookbooks/aws-parallelcluster-environment/files/default/ec2_udev_rules/ec2_volume.py

```diff
@@ -0,0 +1,56 @@
+import configparser
+import re
+
+import requests
+from botocore.config import Config
+
+METADATA_REQUEST_TIMEOUT = 60
+
+
+def get_imdsv2_token():
```

Comment:
Adding this file in the folder is not enough.

This will give you the possibility to experiment with the cookbook. 
If you search for `manageVolume.py` in the cookbook code you can see:

```
  cookbook_file 'manageVolume.py' do
    source 'ec2_udev_rules/manageVolume.py'
    cookbook 'aws-parallelcluster-environment'
    path '/usr/local/sbin/manageVolume.py'
    user 'root'
    group 'root'
    mode '0755'
  end
```

You should do something similar. 

You can start checking:
* `resources/ec2_udev_rules/partial/_common_udev_configuration.rb` --> where we are creating the file in the system
* `spec/unit/resources/ec2_udev_rules_spec.rb` --> unit test you can run with `cd cookbooks/aws-parallelcluster-environment; chef exec rspec ./spec/unit/resources/os_type_spec.rb`
* `test/controls/ec2_udev_rules_spec.rb` --> kitchen tests you can run with `bash kitchen.ec2.sh environment-install test ec2-udev-rules -c 6` --> see README first.

Reply:
You could evaluate to do these changes as part of this patch or avoid to create the new ec2_volume.py for now and just write unit tests even if the code is duplicate. Then in another patch you can work to have this code in common and experiment with cookbook changes to add the new file.

Reply:
I see, I'll just have the duplicate code for now and work on the patch in the future then.